nipap-cli/nipap_cli/nipap_cli.py

```diff
@@ -557,12 +557,17 @@ def add_prefix(arg, opts, shell_opts):
         parent_prefix = args['from-prefix'][0]
         parent_op = 'equals'
     else:
-        parent_prefix = opts.get('prefix').split('/')[0]
+        parent_prefix = opts.get('prefix')
         parent_op = 'contains'
 
-    # prefix must be a CIDR network, ie no bits set in host part, so we
+    # If bits are set in the host part we assume a host is being added and
     # remove the prefix length part of the prefix as then the backend will
     # assume all bits being set
+    try:
+        _ = IPy.IP(parent_prefix)
+    except:
+        parent_prefix = opts.get('prefix').split("/")[0]
```

Comment:
If we, from some reason, would end up here when assigning a prefix from a prefix (which seems pretty unrealistic), the option 'prefix' doesn't exist (and is wrong).